modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v5_1_2/DDMFormInstanceUpgradeProcess.java

```diff
@@ -216,10 +238,16 @@ else if (Objects.equals(
 						"showPartialResultsToRespondents")) {
 
 				fieldValueJSONObject.put(
-					"value", String.valueOf(expirationDate != null));
+					"value", _getValue(convertedPolls, expirationDate));
 
 				hasShowPartialResultsToRespondents = true;
 			}
+			else if (Objects.equals(
+						fieldValueJSONObject.getString("name"), "published") &&
+					 convertedPolls) {
```

Comment:
@rebsilva could you please change to `convertedPolls && Objects.equals(fieldValueJSONObject.getString("name"), "published")`? Thanks!